cloud_foundry_api/datadog_checks/cloud_foundry_api/cloud_foundry_api.py

```diff
@@ -0,0 +1,143 @@
+                if event_guid == self._last_event_guid or int(time.time()) - event_ts > 600:
+                    scroll = False
+                    break
+                if event_ts > self._last_event_ts:
+                    self._last_event_guid = event_guid
+                    self._last_event_ts = event_ts
+                tags = [
+                    f"event_type:{v2_event['entity']['type']}",
+                    f"{v2_event['entity']['actee_type']}_name:{v2_event['entity']['actee_name']}",
+                    f"{v2_event['entity']['actee_type']}_guid:{v2_event['entity']['actee']}",
+                    f"{v2_event['entity']['actor_type']}_name:{v2_event['entity']['actor_name']}",
+                    f"{v2_event['entity']['actor_type']}_guid:{v2_event['entity']['actor']}",
+                    f"space_guid:{v2_event['entity']['space_guid']}",
+                    f"org_guid:{v2_event['entity']['organization_guid']}",
+                ]
+                dd_event = {
+                    "source_type_name": CloudFoundryApiCheck.SOURCE_TYPE_NAME,
+                    "event_type": v2_event["entity"]["type"],
+                    "timestamp": event_ts,
+                    "msg_title": f"Event {v2_event['entity']['type']} happened for {v2_event['entity']['actee_type']} "
+                    f"{v2_event['entity']['actee_name']}",
+                    "msg_text": f"Triggered by {v2_event['entity']['actor_type']} {v2_event['entity']['actor_name']}",
+                    "priority": "normal",
+                    "tags": tags,
+                }
+                # Make sure we don't send duplicate events in case the pagination gets shifted by new events
+                if event_guid not in events:
+                    events[event_guid] = dd_event
+            # Fetch next page if any
+            next_url = payload.get("next_url")
+            if not next_url or not scroll:
+                break
+            res = self.http.get(f"{self._api_url}{next_url}", headers=headers)
+        return events
+
+    def get_events(self):
+        # type: () -> Dict[str: Event]
+        self.get_oauth_token()
+        if self._api_version == "v2":
+            return self.get_events_v2()
+        elif self._api_version == "v3":
+            # self.get_events_v3()
+            # params = {
+            #     "types": self._event_filter,
+            #     "per_page": min(self._per_page, 5000),
+            #     "order_by": "-created_at",
+            # }
+            # res = self.http.get(f"{self._api_url}/v3/audit_events")
+            pass
+        else:
+            self.log.error("Unknown api version `%s`, choose between `v2` and `v3`", self._api_version)
+
+    def check(self, _):
+        # type: (Dict[str, Any]) -> None
+        events = self.get_events()
+        self.count("events.count", len(events))
+        for event in events.values():
+            self.event(event)
```

Comment:
Is there a batching api for this call?

Reply:
it's the agent taking care of the batching. This call just sends the event to the agent collector